packages/SwingSet/tools/run-utils.js

```diff
@@ -2,12 +2,18 @@ import { Fail, q } from '@endo/errors';
 import { kunser } from '@agoric/kmarshal';
 import { makeQueue } from '@endo/stream';
 
-/** @import { ERef } from '@endo/far' */
+/**
+ * @import { ERef } from '@endo/far'
+ * @import { RunPolicy } from '../src/types-external.js'
+ */
+
+/** @typedef {{ provideRunPolicy: () => RunPolicy | undefined }} RunPolicyMaker */
 
 /**
  * @param {import('../src/controller/controller.js').SwingsetController} controller
+ * @param {RunPolicyMaker} [perfTool]
```

Comment:
Attempted future-proofing: `perfTool: RunPolicyMaker` → `harness: RunHarness` (and likewise in packages/boot/tools/supports.ts and the other files)
```suggestion
/** @typedef {{ provideRunPolicy: () => RunPolicy | undefined }} RunHarness */

/**
 * @param {import('../src/controller/controller.js').SwingsetController} controller
 * @param {RunHarness} [harness] provides run policies for the controller
```